### PR TITLE
Check time lock

### PR DESCRIPTION
**Pull request by arvidn**
implement non-wrapping behavior in `check_time_lock()` to enable in the 3.0 hard fork

---

> [!NOTE]
> **High Risk**
> Consensus-critical validation behavior changes (time-lock enforcement) and introduces a new mode that can reject/accept spends differently around integer overflow boundaries.
> 
> **Overview**
> Adds a new `nowrap` flag to `check_time_locks` (Rust + PyO3) to switch relative height/seconds time-lock math from wrapping additions to saturating additions, preventing overflow-based bypasses.
> 
> Updates Python bindings/type stubs and expands both Rust and Python tests to cover `nowrap` vs wrapping behavior, including explicit overflow/wraparound edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1c5117980ccabfc5ce0aab8e97ff530889b80b31. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: crates/chia-consensus/src/check_time_locks.rs [hunk 1]
```diff
@@ -83,12 +112,14 @@ pub fn py_check_time_locks(
     bundle_conds: &OwnedSpendBundleConditions,
     prev_transaction_block_height: u32,
     timestamp: u64,
+    nowrap: bool,
 ) -> PyResult<Option<u32>> {
```
[the remark this answers is not in the document]
Comment (arvidn) [reply crates/chia-consensus/src/check_time_locks.rs:116]: the API incompatibility is an important feature

File: crates/chia-consensus/src/check_time_locks.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ pub fn check_time_locks(
     bundle_conds: &OwnedSpendBundleConditions,
     prev_transaction_block_height: u32,
     timestamp: u64,
```
[the remark this answers is not in the document]
Comment (arvidn) [reply crates/chia-consensus/src/check_time_locks.rs:16]: it's important to ensure that every call site makes an explicit decision of which behavior it wants

**Review by matt-o-how — DISMISSED** [review]

**Review by matt-o-how — APPROVED** [review]
